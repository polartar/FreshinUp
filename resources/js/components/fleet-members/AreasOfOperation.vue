<template>
  <v-card>
    <v-card-title class="grey--text">
      <v-layout
        row
        space-between
        align-center
      >
        <v-flex>
          <h3>Areas of Operation</h3>
        </v-flex>
        <v-flex shrink>
          <v-dialog
            v-model="newOperation"
            max-width="400"
          >
            <template v-slot:activator="{ on }">
              <v-btn
                slot="activator"
                color="primary"
                text
                @click="newOperation = true"
              >
                <v-icon
                  dark
                  left
                >
                  add_circle_outline
                </v-icon>Add new area
              </v-btn>
            </template>
            <v-card>
              <v-divider />
              <v-card-text class="grey--text">
                Coming Soon
              </v-card-text>
              <v-divider />
            </v-card>
          </v-dialog>
        </v-flex>
      </v-layout>
    </v-card-title>
    <v-divider />
    <v-card-text class="primary--text">
      <div>
        <store-area-list
          :store-areas="operations"
        />
      </div>
    </v-card-text>
    <v-divider />
  </v-card>
</template>

<script>
import StoreAreaList from '../stores/StoreAreaList'
export default {
  name: 'AreasOfOperation',
  components: {
    StoreAreaList
  },
  props: {
    operations: {
      type: Array,
      default: () => []
    }
  },
  data () {
    return {
      newOperation: false
    }
  }
}
</script>

<style lang="scss" scoped>
  .ff-fleet-members__operation-text {
    text-align: center;
  }
</style>
