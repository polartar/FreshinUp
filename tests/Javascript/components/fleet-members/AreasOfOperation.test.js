import { mount } from '@vue/test-utils'
import * as Stories from '~/components/fleet-members/AreasOfOperation.stories'

describe('fleet-members/AreasOfOperation', () => {
  describe('Snapshots', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
