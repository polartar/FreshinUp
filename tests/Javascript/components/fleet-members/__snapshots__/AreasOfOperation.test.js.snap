// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fleet-members/AreasOfOperation Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="v-card v-sheet theme--light"
  >
    <div
      class="v-card__title grey--text"
    >
      <div
        class="layout row space-between align-center"
      >
        <div
          class="flex"
        >
          <h3>
            Areas of Operation
          </h3>
        </div>
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="v-card__text primary--text"
    >
      <div
        class="ff-fleet-members__operation-text"
      >
        
      Nothing here yet.
      
        <a
          href="#"
          slot="activator"
        >
          
            Add new item
          
        </a>
        <div
          class="v-dialog__container"
          style="display: inline-block;"
        >
          <div
            class="v-dialog__content"
            style="z-index: 0;"
            tabindex="-1"
          >
            <div
              class="v-dialog"
              style="max-width: 400px; display: none;"
            >
               
              <div
                class="v-card v-sheet theme--light"
              >
                <hr
                  class="v-divider theme--light"
                />
                 
                <div
                  class="v-card__text grey--text"
                >
                  
            Coming Soon
          
                </div>
                 
                <hr
                  class="v-divider theme--light"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <hr
      class="v-divider theme--light"
    />
  </div>
</div>
`;
